src/client/index.ts

```diff
@@ -717,9 +730,91 @@ export class Client {
 		throw new Error(res.errors.join("\n"));
 	}
 
+	getOutletConfigForCurrentOrigin(){
+		let allIssuers = this.tokenStore.getCurrentIssuers();
+		let currentIssuers = [];
+
+		Object.keys(allIssuers).forEach(key => {
+			let issuerConfig = allIssuers[key] as OffChainTokenConfig;
+
+			try {
+
+				if (
+					(new URL(issuerConfig.tokenOrigin)).origin === document.location.origin 
+					// should not be 2 tokens with same origin
+					// && issuerConfig.collectionID == issuer
+				){
+					currentIssuers.push(issuerConfig);
+				} 
+			} catch (err) {
+				logger(2,err);
+				console.log("issuer config error");
+			}
+		})
+
+		if (currentIssuers.length){
+			return currentIssuers[0];
+		}
+		return false;
+	}
+
+	onlySameOrigin(){
+		let allIssuers = this.tokenStore.getCurrentIssuers();
+		let onlySameOriginFlag = true;
+
+		Object.keys(allIssuers).forEach(key => {
+			let issuerConfig = allIssuers[key] as OffChainTokenConfig;
+			let thisOneSameOrigin = false;
+			try {
+				if (
+					(new URL(issuerConfig.tokenOrigin)).origin === document.location.origin 
+				){
+					thisOneSameOrigin = true;
+				} 
+			} catch (err) {
+				logger(2,err);
+				console.log("issuer config error");
+			}
+
+			// if any issuerConfig missing tokenOrigin or something wrong then skip sameOrigin flow
+			if (!thisOneSameOrigin) {
+				onlySameOriginFlag = false;
+			}
+		})
+
+		return onlySameOriginFlag;
+	}
+
 	on(type: TokenNegotiatorEvents, callback?: any, data?: any) {
 		requiredParams(type, "Event type is not defined");
 
+		if(type == 'token-proof') {
+			logger(2, "token-proof listener atteched. check URL HASH for proof callbacks.");
+			let params =
+			window.location.hash.length > 1
+			? "?" + window.location.hash.substring(1)
+			: window.location.search;
+			let urlParams = new URLSearchParams(params);
+			
+			const action = urlParams.get("action");
+						
+			if (action == "proof-callback") {
```

Comment:
Should this be ===

Reply:
fixed in https://github.com/TokenScript/token-negotiator/pull/440/commits/534669300597f5cb437aff3cbd7677c37ce6c4e9